### PR TITLE
fix(oracle): add missing default and not null condition to addColumn

### PR DESCRIPTION
**Pull request by hjamil-24**
## Pull Request Checklist

- [x] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

This PR is to allow default values and not null condition for `BOOLEAN` and `ENUM`.
While adding columns, the options `defaultValue` and `allowNull` have no impact.

Example:
```
queryInterface.addColumn('menu_uploads', 'displayable', {
      type: Sequelize.DataTypes.BOOLEAN(),
      defaultValue: true,
      allowNull: false
});
```
SQL expected: `ALTER TABLE "menu_uploads" ADD "displayable" CHAR(1) DEFAULT 1 NOT NULL CHECK ("displayable" IN('1',  '0'))`

### CODE REVIEW

File: test/unit/sql/add-column.test.js [hunk 1]
```diff
@@ -69,6 +69,28 @@ describe(Support.getTestDialectTeaser('SQL'), () => {
       });
     }
 
+    if (['oracle'].includes(current.dialect.name)) {
```
Comment (WikiRik) [inline test/unit/sql/add-column.test.js:72]: Preferably we have the expectations of other dialects here as well, or is this functionality not available on other dialects?

**Review by WikiRik — COMMENTED** [review]
I'll rerun the CI, much more is failing than due to your change

File: test/unit/sql/add-column.test.js (hunk 1 above)
Comment (hjamil-24) [reply test/unit/sql/add-column.test.js]: I have added expectations for other dialects.

**Comment by hjamil-24** [comment]
> I'll rerun the CI, much more is failing than due to your change

The errors are coming because of `node-gyp` not found. This is the case with other PRs on v6. See [Link](https://github.com/sequelize/sequelize/actions/runs/6324119533/job/17173007868?pr=16388).

**Review by WikiRik — APPROVED** [review]

**Comment by Talal916** [comment]
No chances this affected behavior for mysql right?

**Comment by WikiRik** [comment]
> No chances this affected behavior for mysql right? 

Only the oracle query generator was affected, so this does not affect mysql.

**Pull request merged**